src/ConstructiveSolidGeometry/Transformations.jl

```diff
@@ -1,69 +1,27 @@
-struct ScaledGeometry{T,P<:AbstractGeometry{T}} <: AbstractTransformedGeometry{T}
-    p::P
-    inv_s::SVector{3,T}
-    ScaledGeometry(p::P, s::SVector{3,T}) where {T,P} = new{T,P}(p, inv.(s))
+const Transformations{T} = NamedTuple{(:scale, :rotation, :translation), 
+    Tuple{SVector{3, T}, SMatrix{3, 3, T, 9}, CartesianVector{T}}}
+# Transformation rule:
+#   1. Scale
+#   2. Rotate
+#   3. Translate
+
+transform(g::AbstractPrimitive, t::Transformations) =
+    transform(transform(transform(g, t.scale), t.rotation), t.translation)
+
+transform(p::AbstractVolumePrimitive, v::CartesianVector) = p + v
+transform(p::AbstractVolumePrimitive, r::AbstractMatrix) = r * p
+transform(p::AbstractVolumePrimitive, s::SVector{3}) = s * p
+
+transform(csg::G, t::Transformations) where {G <: AbstractConstructiveGeometry} = G(transform(csg.a, t), transform(csg.b, t))
+
+# Not sure yet if Rotations will always be of type SMatrix{3, 3, T, 9} -> Any
+
+function combine_transformations(current::Transformations, new::Transformations)
+    s = current.scale .* new.scale
+    t = scale(current.translation, new.scale)
+    r = new.rotation * current.rotation    
+    t = new.rotation * t
+    t = t + new.translation
```

Comment:
Is this tested? I don't think this will work properly for scaled geometries. If we rotate a `Box` around 45°, then scale along the new X, Y and Z and rotate 45° back, we would expect scaling along some diagonals.

Reply:
I don't see how to construct primitives like this just by scaling in `x`,`y`,`z`:
![Spat](https://user-images.githubusercontent.com/30291312/121859628-3b132c00-ccf8-11eb-98dd-873f03db1976.png)
(Source: https://de.wikipedia.org/wiki/Spatprodukt)

Reply:
True. For proper scaling we would need to add shear module to the volume primitives. 
As scaling is not that relevant I would postpone that into somewhere in the future. 
I removed scaling, see https://github.com/JuliaPhysics/SolidStateDetectors.jl/pull/163/commits/122e79c65eb785b7e15606ac251c3fea23e07980.